server/services/participants.js

```diff
@@ -6,12 +6,81 @@ const { dbClient, collections } = require('../db');
 const { createRows, verifyHeaders } = require('../utils');
 const { userRegionQuery } = require('./user.js');
 
-const getParticipants = async (user) => {
+const setParticipantStatus = async (employerId, participantId, status) => {
+  await dbClient.db[collections.PARTICIPANTS_STATUS].update({
+    employer_id: employerId,
+    participant_id: participantId,
+    current: true,
+  }, { current: false });
+  return dbClient.db[collections.PARTICIPANTS_STATUS].save({
+    employer_id: employerId,
+    participant_id: participantId,
+    status,
+    current: true,
+  });
+};
+
+const flatJoinedParticipantStatus = (raw) => {
+  const prticipantsMap = new Map();
+  const participantsStatusMap = new Map();
+
+  raw.forEach((item) => {
+    prticipantsMap.set(item.participants__id, {
+      ...item.participants__body,
+      id: item.participants__id,
+    });
+
+    const participantStatus = participantsStatusMap.get(item.participants_status__participant_id);
+
+    participantsStatusMap.set(
+      item.participants_status__participant_id,
+      participantStatus ? [...participantStatus, item] : [item],
+    );
+  });
+
+  const participants = [];
+
+  prticipantsMap.forEach((item, participantId) => {
```

Comment:
Is there no way to do a join? The current implementation performs a query for each participant, right?

Reply:
Ignore this. I was mistaken.

Reply:
The join is in line 67. This function is necessary because we can have repeated participants in the output due to the combination with the `participants_status` table. For each `participantId` we have a join with the participant table, and therefore an employer can engage multiple participants.